Week-3/Exercise3.4/index.js

```diff
@@ -0,0 +1,45 @@
+function createStack() {
+  return {
+    items: [],
+    push(item) {
+      this.items.push(item);
+    },
+    pop() {
+      return this.items.pop();
+    }
+  };
+}
+
+const stack = createStack();
+
+stack.push(10);
+stack.push(5);
+stack.pop();// => 5
+stack.items;// => [10]
+stack.items = [10, 100, 1000];// Encapsulation broken!
+
+// In the above example stack.items is resulting 10 because we are accessing items array directly and thus we can read its value
+
+
+//new implementation
+function createStack() {
+  const items = [];
+
+  return {
+    push(item) {
```

Comment:
we can just return the reference of those functions in return as object ,avoid adding logics in return statements in such scenarios

Reply:
Refactored the logic